sktime/forecasting/compose/_dmlf.py

```diff
@@ -0,0 +1,413 @@
+        else:
+            time_idx = y.index
+
+        insample_fh = ForecastingHorizon(time_idx, is_relative=False)
+        y_pred_insample = self.forecaster_y_insample_.predict(fh=insample_fh, X=X_conf)
+        self.y_residuals_ = y - y_pred_insample
+
+        # Step 2: Fit exposure model for residuals
+        self.forecaster_ex_insample_ = clone(self.forecaster_ex)
+        self.forecaster_ex_insample_.fit(X_ex, X=X_conf, fh=fh)
+
+        # Get in-sample exposure residuals
+        X_ex_pred_insample = self.forecaster_ex_insample_.predict(
+            fh=insample_fh, X=X_conf
+        )
+        self.X_ex_residuals_ = X_ex - X_ex_pred_insample
+
+        # Step 3: Fit residual relationship
+        self.forecaster_res_ = clone(self.forecaster_res)
+        self.forecaster_res_.fit(self.y_residuals_, X=self.X_ex_residuals_, fh=fh)
+
+        # Step 4: Fit final versions for prediction
+        self.forecaster_y_final_ = clone(self.forecaster_y)
+        self.forecaster_y_final_.fit(y, X=X_conf, fh=fh)
+
+        self.forecaster_ex_final_ = clone(self.forecaster_ex)
+        self.forecaster_ex_final_.fit(X_ex, X=X_conf, fh=fh)
+
+        return self
+
+    def _predict(self, fh=None, X=None):
+        """Generate DML forecasts.
+
+        Prediction logic:
+        1. Split X_new into exposure and confounder variables
+        2. Get base forecast from confounders: y_pred_conf
+        3. Get de-confounded exposure: X_ex_deconf
+        4. Get causal forecast component: y_pred_causal
+        5. Return combined forecast: y_pred_conf + y_pred_causal
+        """
+        # Split exogenous data
+        X_ex_new, X_conf_new = self._split_exogenous_data(X)
+
+        # Base forecast from confounders
+        y_pred_conf = self.forecaster_y_final_.predict(fh=fh, X=X_conf_new)
+
+        # De-confounded exposure
+        X_ex_pred = self.forecaster_ex_final_.predict(fh=fh, X=X_conf_new)
+        X_ex_deconf = X_ex_new - X_ex_pred
+
+        # Causal forecast component
+        y_pred_causal = self.forecaster_res_.predict(fh=fh, X=X_ex_deconf)
+
+        # Combined forecast
+        return y_pred_conf + y_pred_causal
+
+    def _predict_interval(self, fh, X=None, coverage=0.9):
+        """Generate prediction intervals for DML forecasts."""
+        # Split exogenous data
+        X_ex_new, X_conf_new = self._split_exogenous_data(X)
+
+        # Base prediction intervals
+        pred_int_conf = self.forecaster_y_final_.predict_interval(
+            fh=fh, X=X_conf_new, coverage=coverage
+        )
+
+        # De-confounded exposure
+        X_ex_pred = self.forecaster_ex_final_.predict(fh=fh, X=X_conf_new)
+        X_ex_deconf = X_ex_new - X_ex_pred
+
+        # Causal prediction intervals
+        pred_int_causal = self.forecaster_res_.predict_interval(
+            fh=fh, X=X_ex_deconf, coverage=coverage
+        )
+
+        # Combine intervals (assumes independence)
+        return pred_int_conf + pred_int_causal
```

Comment:
I am not sure if this would be as straightforward as adding these two predictions. Depending on the estimator used, both of the quantities would have some posterior distribution, and the final value would depend on the distributions.